Add datatype validation

Currently, it is possible to use variables of any datatype to map them to message fields. This will eventually result in an error message on the rosbridge:

Reproduce the error:
- Create a publisher on a topic with e.g. `geometry_msgs/Pose`
- create variable with value `0`
- select this variable for `frame_id(string)`
- Result: `[ERROR] [1618848091.913749]: [Client 7] publish: geometry_msgs/PoseStamped message requires a string for field header.frame_id, but got a <type 'int'>`

We should add a validation step on URCap side.

As far as I know, polyscope variables are not typed.

We should try to convert the variables to string / numbers in URScript to get a type check at runtime. Currently, we convert everything to a string, e.g. `to_str(int_var)`. As this integer ends up in the jsonstring without quotes it will be treated as number on the rosbridge side. We'll have to make sure that the jsonstring actually contains surrounding quotes for a string variable and tries to convert it `to_num` for numeric values.

--- a/src/main/java/de/fzi/ros_as_a_service/impl/RosTaskProgramSuperNodeContribution.java
+++ b/src/main/java/de/fzi/ros_as_a_service/impl/RosTaskProgramSuperNodeContribution.java
@@ -743,7 +743,7 @@ public abstract class RosTaskProgramSuperNodeContribution implements ProgramNode
     // replace string variables
     result = result.replaceAll(
         "\\{\\\" \\+ quote \\+ \\\"-\\+useVar\\+-\\\" \\+ quote \\+ \\\":\\\" \\+ quote \\+ \\\"([^\\\"]*)\\\" \\+ quote \\+ \\\"\\}",
-        "\\\" + quote + $1 + quote + \\\"");
+        "\\\" + quote + to_str($1) + quote + \\\"");
     // System.out.println("without useVar : " + result);
 
     // replace numeric variables

